services/state/addressbook/node.proto

```diff
@@ -0,0 +1,115 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "basic_types.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * Representation of a Node in the network Merkle tree
+ *
+ * As with all network entities, a Node has a unique nodeId, which is usually given along
+ */
+
+message Node{
+    /**
+     * The unique entity id of the DynamicAddress.
+     */
+    int64 node_id = 1;
+
+    /**
+     * The account to be paid for queries and transactions sent to this node
+     */
+    AccountID account_id = 2;
+
+    /**
+     * A description of the node, with UTF-8 encoding up to 100 bytes
+     */
+    string description = 4;
+
+    /**
+    * Node Gossip inbound Endpoints, ip address or FQDN and port
+    */
+    ServiceEndpoint inbound_gossip_endpoint = 5;
+
+    /**
+     * Node Gossip outbound Endpoints, ip address or FQDN and port
+     */
+    ServiceEndpoint outbound_gossip_endpoint = 6;
+
+    /**
+    * # A node's service Endpoints, ip address or FQDN and port
+    */
+    repeated ServiceEndpoint service_endpoint = 7;
+
+    /**
+    * node’s public key within the Address Book used for signing.
+     * The node's X509 RSA public key used to sign stream files (e.g., record stream
+     * files). Precisely, this field is a string of hexadecimal characters which,
+     * translated to binary, are the public key's DER encoding.
+    */
```

Comment:
```suggestion
    /**
     * node’s public key within the Address Book used for signing.
     * The node's X509 RSA public key used to sign stream files (e.g., record stream
     * files). Precisely, this field is a string of hexadecimal characters which,
     * translated to binary, are the public key's DER encoding.
    */
```

Is it guaranteed to be specified? (In the Create transaction, it says this is optional, I think. I don't understand how that can be). Is it guaranteed to be always valid?

Reply:
public_key is mandatory; certificate_hash is optional.